specification/operationalinsights/resource-manager/Microsoft.OperationalInsights/preview/2020-03-01-preview/Tables.json

```diff
@@ -0,0 +1,216 @@
+      "get": {
+        "tags": [
+          "Tables"
+        ],
+        "x-ms-examples": {
+          "TablesGet": {
+            "$ref": "./examples/TablesList.json"
+          }
+        },
+        "operationId": "Tables_ListByWorkspace",
+        "description": "Gets all the tables for the specified Log Analytics workspace.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK response definition.",
+            "schema": {
+              "$ref": "#/definitions/TablesListResult"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}/providers/Microsoft.OperationalInsights/workspaces/{workspaceName}/tables/{tableName}": {
+      "put": {
+        "tags": [
+          "Tables"
+        ],
+        "x-ms-examples": {
+          "TablesSet": {
+            "$ref": "./examples/TablesSet.json"
+          }
+        },
+        "operationId": "Tables_Set",
+        "description": "Sets a Log Analytics workspace table properties.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "../../../common/v1/types.json#/parameters/WorkspaceNameParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/TableNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/Table"
+            },
+            "description": "The parameters required to set table properties."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK response definition.",
+            "schema": {
+              "$ref": "#/definitions/Table"
+            }
+          }
+        }
+      },
+      "get": {
```

Comment:
don't we have a delete method?

Reply:
Tables cannot be added/deleted by a user, the list is managed by features added/removed to/from the workspace elsewhere. The idea is to allow a user to changes properties of such data table , e.g. retention in days.

Reply:
The PUT is to update, not create. The tables are managed at the backend by the service. A customer may add/remove features to/from their workspace which may add/remove tables.

Reply:
So, we have a Log analytics workspace resource, which is kind of a container for data. The workspace have tables. Those are like a data table in Kusto. Data is ingested into those tables, and a customer can query the data using Kusto.
Some tables are built in. Some are created/enabled when a customer enables solutions on the workspace. Some tables are created when a customer sends custom data to a collection endpoint.
This API enables GET/PUT/LIST for tables, so a customer can set table properties - for now we have just the retention in days.
Becasue the API does not manage the tables lifetime, we don't provide DELETE and the PUT  is actually a 'set' command and not a 'Create'.

Reply:
Ideally,in that case, this update should be allowed via a PATCH and not PUT. The current PUT behavior violates the definition of PUT which is create or update. 
 how are names of these generated? how would the customer know which table he needs to update? How would be get to the individual table from the list of tables? why do you want to expose this a sub resource type? It feels like providing half baked management when PUT is not present but GET/LIST/PATCH are. Can it be exposed as a property on the workspace whose retention can be updated.

Reply:
@yoramsinger please take a look at above comment